macaroons/bake_test.go

```diff
@@ -0,0 +1,54 @@
+package macaroons_test
+
+import (
+	"context"
+	"encoding/hex"
+	"testing"
+
+	"github.com/lightningnetwork/lnd/macaroons"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc/metadata"
+	"gopkg.in/macaroon-bakery.v2/bakery"
+)
+
+// TestBakeFromRootKey tests that a macaroon can be baked from a root key
+// directly without needing to create a store or service first.
+func TestBakeFromRootKey(t *testing.T) {
+	// Create a test store and unlock it.
+	_, store := newTestStore(t)
+
+	pw := []byte("weks")
+	err := store.CreateUnlock(&pw)
+	require.NoError(t, err)
+
+	// Force the store to create a new random root key.
+	key, _, err := store.RootKey(defaultRootKeyIDContext)
+	require.NoError(t, err)
+	require.Len(t, key, 32)
```

Comment:
nit:
Could also add test coverage of the `store.Get` function here, with something like
```suggestion
	key, id, err := store.RootKey(defaultRootKeyIDContext)
	require.NoError(t, err)
	require.Len(t, key, 32)

	tmpKey, err := store.Get(defaultRootKeyIDContext, id)
	require.NoError(t, err)
	require.Equal(t, key, tmpKey)
```